### PR TITLE
feat: add WatcherEX interface

### PR DESCRIPTION
**Pull request by Shivansh-yadav13**
Signed-off-by: Shivansh-yadav13 <yadavshivansh@gmail.com>

### CODE REVIEW

**Comment by Shivansh-yadav13** [comment]
@nodece pls review

File: src/persist/watcherEX.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { Watcher } from './watcher';
+import { Model } from '../model';
+
+export interface WatcherEX extends Watcher {
+  UpdateForAddPolicy(sec: string, ptype: string, ...params: string[]): void;
```
Comment (nodece) [inline src/persist/watcherEX.ts]:
```suggestion
  UpdateForAddPolicy(sec: string, ptype: string, ...params: string[]): Promise<void>;
```

File: src/persist/watcherEX.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+  UpdateForRemovePolicy(sec: string, ptype: string, ...params: string[]): void;
```
Comment (nodece) [inline src/persist/watcherEX.ts]:
```suggestion
  UpdateForRemovePolicy(sec: string, ptype: string, ...params: string[]): Promise<void>;
```

File: src/persist/watcherEX.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+  UpdateForRemoveFilteredPolicy(sec: string, ptype: string, fieldIndex: number, ...fieldValues: string[]): void;
```
Comment (nodece) [inline src/persist/watcherEX.ts]:
```suggestion
  UpdateForRemoveFilteredPolicy(sec: string, ptype: string, fieldIndex: number, ...fieldValues: string[]): Promise<void>;
```

File: src/persist/watcherEX.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+  UpdateForSavePolicy(model: Model): void;
```
Comment (nodece) [inline src/persist/watcherEX.ts]:
```suggestion
  UpdateForSavePolicy(model: Model): Promise<void>;
```

File: src/persist/watcherEX.ts [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+  UpdateForAddPolicies(sec: string, ptype: string, ...rules: string[][]): void;
```
Comment (nodece) [inline src/persist/watcherEX.ts]:
```suggestion
  UpdateForAddPolicies(sec: string, ptype: string, ...rules: string[][]): Promise<void>;
```

File: src/persist/watcherEX.ts [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+
+  UpdateForRemovePolicies(sec: string, ptype: string, ...rules: string[][]): void;
```
Comment (nodece) [inline src/persist/watcherEX.ts]:
```suggestion
  UpdateForRemovePolicies(sec: string, ptype: string, ...rules: string[][]): Promise<void>;
```

File: src/persist/watcherEX.ts [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { Watcher } from './watcher';
+import { Model } from '../model';
+
+export interface WatcherEX extends Watcher {
```
Comment (nodece) [inline src/persist/watcherEX.ts]:
```suggestion
export interface WatcherEx extends Watcher {
```

Comment (Shivansh-yadav13) [reply src/persist/watcherEX.ts]: Done :+1:

**Review by nodece — APPROVED** [review]

**Pull request merged**